common/src/main/java/com/wynntils/sockets/SocketClientHandler.java

```diff
@@ -0,0 +1,135 @@
+    }
+
+    @Override
+    public void onDisconnect() {
+        WynntilsMod.postEvent(new SocketEvent.Disconnected());
+
+        if (WorldStateManager.onServer()) {
+            McUtils.sendMessageToClient(
+                    new TextComponent("Disconnected from HadesServer").withStyle(ChatFormatting.RED));
+        }
+
+        WynntilsMod.info("Disconnected from HadesServer.");
+
+        HadesUserModel.getHadesUserMap().clear();
+    }
+
+    @Override
+    public void handleAuthenticationResponse(HSPacketAuthenticationResponse packet) {
+        Component userComponent = TextComponent.EMPTY;
+
+        switch (packet.getResponse()) {
+            case SUCCESS -> {
+                WynntilsMod.info("Successfully connected to HadesServer: " + packet.getMessage());
+                userComponent =
+                        new TextComponent("Successfully connected to HadesServer").withStyle(ChatFormatting.GREEN);
+                WynntilsMod.postEvent(new SocketEvent.Authenticated());
+            }
+            case INVALID_TOKEN -> {
+                WynntilsMod.error("Got invalid token when trying to connect to HadesServer: " + packet.getMessage());
+                userComponent = new TextComponent("Got invalid token when connecting HadesServer")
+                        .withStyle(ChatFormatting.RED);
+            }
+            case ERROR -> {
+                WynntilsMod.error("Got an error trying to connect to HadesServer: " + packet.getMessage());
+                userComponent =
+                        new TextComponent("Got error when connecting HadesServer").withStyle(ChatFormatting.RED);
+            }
+        }
+
+        if (WorldStateManager.onServer()) {
+            McUtils.sendMessageToClient(userComponent);
+        }
+    }
+
+    @Override
+    public void handlePing(HSPacketPong packet) {
+        // noop at the moment
+        // todo eventually calculate ping
+    }
+
+    @Override
+    public void handleUpdateMutual(HSPacketUpdateMutual packet) {
+        if (!SocketFeature.INSTANCE.getOtherPlayerInfo) return;
+
+        Optional<HadesUser> userOptional = HadesUserModel.getUser(packet.getUser());
+        if (userOptional.isPresent()) {
+            userOptional.get().updateFromPacket(packet);
+        } else {
+            HadesUserModel.putUser(packet.getUser(), new HadesUser(packet));
+        }
+    }
+
+    @Override
+    public void handleDiscordLobbyServer(HSPacketDiscordLobbyServer packet) {
+        // noop for now
+    }
+
+    @Override
+    public void handleClearMutual(HSPacketClearMutual packet) {
+        HadesUserModel.removeUser(packet.getUser());
+    }
+
+    @Override
+    public void handleDisconnect(HSPacketDisconnect packet) {
+        WynntilsMod.info("Disconnected from HadesServer. Reason: " + packet.getReason());
+
+        if (WorldStateManager.onServer()) {
+            McUtils.sendMessageToClient(new TextComponent("[Wynntils/Artemis] Disconnected from HadesServer.")
```

Comment:
Inconsistent format, others do not use prefix

Reply:
We do use this elsewhere

Reply:
Update reminder does this too.